### ISSUE TITLE
Open API genrated seems to ignore nested array objects while updgrading from 2.1.0 to 2.5.0

### ISSUE THREAD

**Issue by srikarrampa** [springdoc/springdoc-openapi#2562]
**Describe the bug**

- If you are reporting a bug, please help to speed up problem diagnosis by providing as
  much information as possible:
- A clear and concise description of what the bug is: the title of an issue is not enough

**To Reproduce**

- What version of spring-boot you are using? 3.2.4
- What modules and versions of springdoc-openapi are you using? 2.5.0
- What is the actual and the expected result using OpenAPI Description (yml or json)?
```
public class TestObject implements Serializable {
  private Type type;
  private Value value;
  private List<TestObject> args;
```

openapi in 2.1.0
```
"TestObject" : {
        "properties" : {
          "args" : {
            "items" : {
              "$ref" : "#/components/schemas/TestObject"
            },
            "type" : "array"
          },
          "type" : {
            "enum" : [ "value1", "value2"],
            "type" : "string"
          },
          "value" : {
            "$ref" : "#/components/schemas/Value"
          }
        },
        "type" : "object"
      },
```
openapi in 2.5.0

```
"TestObject" : {
        "properties" : {
          "type" : {
            "enum" : [ "value1", "value2"],
            "type" : "string"
          },
          "value" : {
            "$ref" : "#/components/schemas/Value"
          }
        },
        "type" : "object"
      },
```

- Provide with a sample code (HelloController) or Test that reproduces the problem

**Expected behavior**

- A clear and concise description of what you expected to happen.
Upgrade to 2.5.0 shouldn't remove the nested array from the schema

**Screenshots**

**Additional context**

**Comment by bnasslahsen** [comment]
@srikarrampa,

Not reproducible.
Feel free to provide a [Minimal, Reproducible Example - with HelloController](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the problem.

This ticket will be closed, but can be reopened if your provide the reproducible sample.

**Comment by srikarrampa** [comment]
Thanks for looking into the issue @bnasslahsen. I tried reproduce the with minimal example and couldn't reproduce the same.  ie., I tried to create a controller with reponse object as TestObject as present above. Didn't notice a array object disappearing.

However in the actual usecase, the TestObject is nested within multiple objects. 

One more thing I did try is changing the List interface -> ArrayList and the openapi spec was perfect. Didn't see array object disappearing.

public class TestObject implements Serializable {
  private Type type;
  private Value value;
  private ArrayList<TestObject> args;
}

"TestObject" : {
        "properties" : {
          "args" : {
            "items" : {
              "$ref" : "#/components/schemas/TestObject"
            },
            "type" : "array"
          },
          "type" : {
            "enum" : [ "value1", "value2"],
            "type" : "string"
          },
          "value" : {
            "$ref" : "#/components/schemas/Value"
          }
        },
        "type" : "object"
      },
